src/weather_data.cpp

```diff
@@ -180,9 +99,45 @@ bool acidic( weather_type const type )
 {
     return weather_data_internal( type ).datum.acidic;
 }
-weather_effect_fn effect( weather_type const type )
+std::vector<std::pair<std::string, int>> effects( weather_type const type )
+{
+    return weather_data_internal( type ).datum.effects;
+}
+std::string tiles_animation( weather_type const type )
+{
+    return weather_data_internal( type ).datum.tiles_animation;
+}
+weather_animation_t get_weather_animation( weather_type const type )
+{
+    return weather_data_internal( type ).datum.weather_animation;
+}
+int sound_category( weather_type const type )
```

Comment:
Generally: instead of having all those small functions, you could expose a (constant) reference to the weather data object and let the caller get the desired property from it.

```C++
const weather_datum &weather::data( weather_type const type ) { return weather_data_internal( type ).datum; }

// Caller does instead of 
x = get_weather_animation( t );
// this:
x = weather::data( t ).weather_animation;
```

That way you don't have to add a new function for each property of the weather datum.